vt/client.py

```diff
@@ -973,3 +975,60 @@ async def _wait_for_analysis_completion(self, analysis: Object) -> Object:
 
   async def wait_for_analysis_completion(self, analysis: Object) -> Object:
     return await self._wait_for_analysis_completion(analysis)
+
+
+  def scan_file_private(
+      self, 
+      file: typing.Union[typing.BinaryIO, str],
+      code_insight: bool = False,
+      wait_for_completion: bool = False
+  ) -> Object:
+      """Scan file privately with optional code insight analysis.
+      
+      Args:
+          file: File to scan (path string or file object)
+          code_insight: Enable code analysis features
+          wait_for_completion: Wait for completion
+          
+      Returns:
+          Object: Analysis object with scan results
+      """
+      return make_sync(
+          self.scan_file_private_async(file, code_insight, wait_for_completion)
+      )
+
+  async def scan_file_private_async(
+      self,
+      file: typing.Union[typing.BinaryIO, str],
+      code_insight: bool = False, 
+      wait_for_completion: bool = False
+  ) -> Object:
+      """Async version of scan_file_private"""
+
+      # Handle string path
+      if isinstance(file, str):
+          async with aiofiles.open(file, 'rb') as f:
+              file_content = io.BytesIO(await f.read())
+              file_content.name = os.path.basename(file)
+              return await self.scan_file_private_async(
+                  file_content,
+                  code_insight=code_insight,
+                  wait_for_completion=wait_for_completion
+              )
+
+      # Create form data
+      form = aiohttp.FormData()
+      form.add_field('file', file)
+      form.add_field('private', '1')
+      form.add_field('code_insight', '1' if code_insight else '0')
+
+      # Get upload URL and submit file
+      upload_url = await self.get_data_async("/files/upload_url")
```

Comment:
That URL would upload the files as public, this is the correct URL:
```suggestion
      upload_url = await self.get_data_async("/private/files/upload_url")
```